examples/ex24p.cpp

```diff
@@ -0,0 +1,334 @@
+
+   a->AddDomainIntegrator(new VectorFEMassIntegrator(*sigma));
+
+   a_NDH1->AddDomainIntegrator(new MixedVectorGradientIntegrator(*muinv));
+
+   // 9. Assemble the parallel bilinear form and the corresponding linear
+   //    system, applying any necessary transformations such as: parallel
+   //    assembly, eliminating boundary conditions, applying conforming
+   //    constraints for non-conforming AMR, static condensation, etc.
+   if (static_cond) { a->EnableStaticCondensation(); }
+
+   a->Assemble();
+   if (!pa) { a->Finalize(); }
+
+   a_NDH1->Assemble();
+   if (!pa) { a_NDH1->Finalize(); }
+
+   Vector B(fespace->GetTrueVSize());
+   Vector X(fespace->GetTrueVSize());
+
+   if (pa)
+   {
+      a_NDH1->Mult(p, x);
+      x.GetTrueDofs(B);
+   }
+   else
+   {
+      HypreParMatrix *NDH1 = a_NDH1->ParallelAssemble();
+
+      Vector P(H1fespace->GetTrueVSize());
+      p.GetTrueDofs(P);
+
+      NDH1->Mult(P,B);
+
+      delete NDH1;
+   }
+
+   // 10. Define and apply a parallel PCG solver for AX=B with no
+   //     preconditioner, in the full assembly case.
+   //     With partial assembly, use Jacobi preconditioner.
+
+   if (pa)
+   {
+      ParGridFunction diag_pa(fespace);
+      Vector tdiag_pa(fespace->GetTrueVSize());
+      a->AssembleDiagonal(tdiag_pa);
+      diag_pa.SetFromTrueDofs(tdiag_pa);
+
+      Array<int> ess_tdof_list;
+      OperatorJacobiSmoother Jacobi(diag_pa, ess_tdof_list, 1.0);
+
+      CGSolver cg(MPI_COMM_WORLD);
+      cg.SetRelTol(1e-12);
+      cg.SetMaxIter(1000);
+      cg.SetPrintLevel(1);
+      cg.SetOperator(*a);
+      cg.SetPreconditioner(Jacobi);
+
+      ParGridFunction rhs(fespace);
+      rhs = x;
+      x = 0.0;
+      cg.Mult(rhs, x);
+   }
+   else
+   {
+      HypreParMatrix *Amat = a->ParallelAssemble();
+      HyprePCG pcg(*Amat);
+      pcg.SetTol(1e-12);
+      pcg.SetMaxIter(1000);
+      pcg.SetPrintLevel(2);
+      pcg.Mult(B, X);
+
+      delete Amat;
+
+      x.SetFromTrueDofs(X);
+   }
+
+   // 11. Compute the same solution by applying GradientInterpolator in H(curl).
+
+   ParDiscreteLinearOperator grad(H1fespace, fespace);
+   grad.AddDomainInterpolator(new GradientInterpolator());
+   grad.Assemble();
+
+   ParGridFunction gradp(fespace);
+   grad.Mult(p, gradp);
+
+   // 12. Compute the projection of the exact grad p.
+
+   ParGridFunction exact_gradp(fespace);
+   exact_gradp.ProjectCoefficient(gradp_coef);
+
+   // 13. Compute and print the L^2 norm of the error.
+   {
+      double errSol = x.ComputeL2Error(gradp_coef);
+      double errInterp = gradp.ComputeL2Error(gradp_coef);
+      double errProj = exact_gradp.ComputeL2Error(gradp_coef);
+
+      if (myid == 0)
+      {
+         cout << "\n Solution of (E_h,v) = (grad p,v) for E_h and v in H(curl)"
+              ": || E_h - grad p ||_{L^2} = " << errSol << '\n' << endl;
+         cout << " Gradient interpolant E_h = grad p in H(curl): || E_h - grad"
+              "p ||_{L^2} = " << errInterp << '\n' << endl;
```

Comment:
The way the string is split here to a new line, there is no space between "grad" and "p".